script_test.go

```diff
@@ -1850,6 +1850,104 @@ func TestReadReturnsErrorGivenReadErrorOnPipe(t *testing.T) {
 	}
 }
 
+var base64Cases = []struct {
+	name    string
+	decoded string
+	encoded string
+}{
+	{
+		name:    "empty string",
+		decoded: "",
+		encoded: "",
+	},
+	{
+		name:    "single line string",
+		decoded: "hello world",
+		encoded: "aGVsbG8gd29ybGQ=",
+	},
+	{
+		name:    "multi line string",
+		decoded: "hello\nthere\nworld\n",
+		encoded: "aGVsbG8KdGhlcmUKd29ybGQK",
+	},
+}
+
+func TestEncodeBase64(t *testing.T) {
+	t.Parallel()
+
+	for _, tc := range base64Cases {
+		t.Run(tc.name, func(t *testing.T) {
+			got, err := script.Echo(tc.decoded).EncodeBase64().String()
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if got != tc.encoded {
+				t.Errorf("want %s, got %s", tc.encoded, got)
+			}
+		})
+	}
+}
+
+func TestDecodeBase64(t *testing.T) {
+	t.Parallel()
+
+	for _, tc := range base64Cases {
+		t.Run(tc.name, func(t *testing.T) {
+			got, err := script.Echo(tc.encoded).DecodeBase64().String()
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if got != tc.decoded {
+				t.Errorf("want %s, got %s", tc.decoded, got)
+			}
+		})
+	}
+}
+
+func TestEncodeAndDecodeBase64(t *testing.T) {
+	t.Parallel()
+	for _, tc := range base64Cases {
+		t.Run(tc.name, func(t *testing.T) {
+			// encode then decode
+			got, err := script.Echo(tc.decoded).EncodeBase64().DecodeBase64().String()
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if got != tc.decoded {
+				t.Errorf("want %s, got %s", tc.decoded, got)
+			}
+
+			// decode then encode
+			got, err = script.Echo(tc.encoded).DecodeBase64().EncodeBase64().String()
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if got != tc.encoded {
+				t.Errorf("want %s, got %s", tc.encoded, got)
+			}
+		})
+	}
+}
+
+func TestEncodeAndDecodeBase64Bytes(t *testing.T) {
+	t.Parallel()
+	want := []byte{8, 0, 0, 16}
+	input := bytes.NewReader(want)
+
+	got, err := script.NewPipe().WithReader(input).EncodeBase64().DecodeBase64().Bytes()
```

Comment:
We already have some round-trip tests (great!), so what about testing the expected encoded string produced by your byte slice? Use some out-of-band method to produce this (the `base64` tool, for example).